File: t-test/t-test.R
# t-tests

# one sample: sample mean vs known value, null hypothesis means are equal
one_sample_t_test <- function(x,y) {
  mean_x <- sum(x)/length(x)
  variance_x <- sum((x - mean_x)^2)/(length(x)-1)
  t_value <- (mean_x - y) / (sqrt(variance_x) / sqrt(length(x)))
  t_value
}

File: ttest/ttest.py

def one_sample_ttest(x, y):
    """one sample t test
    x = list
    y = single value
    return = t_value as float"""
    mean_x = sum(x)/float(len(x))
    variance_x = sum(([(i - mean_x)**2 for i in x]))/float(len(x) - 1)
    t_value = (mean_x - y) / float((variance_x**(0.5)/ len(x)**(0.5)))
    return t_value
